addons/mrp/models/stock_move.py

```diff
@@ -97,10 +97,24 @@ def _get_aggregated_product_quantities(self, **kwargs):
         """
         aggregated_move_lines = super()._get_aggregated_product_quantities(**kwargs)
         kit_name = kwargs.get('kit_name')
-        if kit_name:
-            for aggregated_move_line in aggregated_move_lines:
-                if aggregated_move_lines[aggregated_move_line]['description'] == kit_name:
+
+        to_be_removed = []
+        for aggregated_move_line in aggregated_move_lines:
+            bom_lines = aggregated_move_lines[aggregated_move_line]['product'].bom_line_ids
+            bom_type = bom_lines[0].bom_id.type if bom_lines else ""
+            description = aggregated_move_lines[aggregated_move_line]['description']
+
+            if kit_name:
+                if bom_type != 'phantom':
+                    to_be_removed.append(aggregated_move_line)
+                elif description == kit_name:
                     aggregated_move_lines[aggregated_move_line]['description'] = ""
+            elif not kwargs and bom_type == 'phantom':
```

Comment:
why `not kwargs?` the other things that can be kwargs seems to be `except_package` and `strict` but if these option are set or not, I'm not sure why this should change the behavior

Reply:
When removing "not kwargs", the test_subkit_in_delivery_slip doesn't pass because _get_aggregated_product_quantities removes everything in aggregated_move_lines because all lines are kit, but kit_name is None